plugin/helpers_bots.go

```diff
@@ -72,3 +91,36 @@ func (p *HelpersImpl) EnsureBot(bot *model.Bot) (retBotId string, retErr error)
 
 	return createdBot.UserId, nil
 }
+
+func (p *HelpersImpl) EnsureBotWithOptions(bot *model.Bot, setters ...Option) (retBotId string, retErr error) {
+	// Default options
+	args := &Options{
+		ProfileImagePath: "",
+		IconImagePath:    "",
+	}
+	for _, setter := range setters {
+		setter(args)
+	}
+	botId, err := p.EnsureBot(bot)
+
+	if err != nil {
+		return botId, err
+	}
+
+	if len(args.ProfileImagePath) > 0 {
```

Comment:
I'm 3/5 that it's more idiomatic to write
```suggestion
	if !args.ProfileImagePath == "" {
```